### ISSUE TITLE
Remove unused inputs

### ISSUE THREAD

**Issue by aidanheerdegen** [ACCESS-NRI/access-esm1.5-configs#13]
It appears the geothermal heating isn't turned on

https://github.com/ACCESS-NRI/access-esm1.5-configs/blob/pre-industrial/ocean/input.nml#L53

Should we remove the `geothermal_heating.nc` input file?

**Comment by aidanheerdegen** [comment]
Also `basin_mask.nc` isn't a model input and was deleted from the OM2 1 degree configurations

https://github.com/COSIMA/access-om2/issues/205

**Comment by aidanheerdegen** [comment]
`CABLE-AUX-1.4/core/biogeochem/pftlookup_csiro_v16_17tiles.csv` is commented out for the historical configuration:

https://github.com/ACCESS-NRI/access-esm1.5-configs/blob/pre-industrial/atmosphere/cable.nml#L56

**Comment by aidanheerdegen** [comment]
`pftlookup_csiro_v16_17tiles_spinup.csv` isn't referenced at all. Neither is `poolcnpInTumbarumba.csv`

**Comment by aidanheerdegen** [comment]
`cf_name_table.txt` seems superfluous. As does `SOURCES` which just contains a bunch of paths.

**Comment by aidanheerdegen** [comment]
We need some input from @MartinDix or Tilo on the CABLE inputs (or @ccarouge?). 

@dougiesquire can you confirm I'm correct about the [geothermal heating](https://github.com/ACCESS-NRI/access-esm1.5-configs/issues/13#issue-2301838946) and [basin mask](https://github.com/ACCESS-NRI/access-esm1.5-configs/issues/13#issuecomment-2116650294)

**Comment by dougiesquire** [comment]
> @dougiesquire can you confirm I'm correct about the https://github.com/ACCESS-NRI/access-esm1.5-configs/issues/13#issue-2301838946 and https://github.com/ACCESS-NRI/access-esm1.5-configs/issues/13#issuecomment-2116650294

Yup, looks right to me (for the `pre-industrial` config at least)

**Comment by ccarouge** [comment]
@aidanheerdegen What do you mean by removing the input files? I guess I don't know what is happening around the ESM1.5 preparations so it's not clear what you mean. 

For the pftlookup file, the namelist file points to a different file, CABLE only uses 1 of these files per configuration so it's ok to remove other pftlookup files.

For the pool file, I'm surprised there isn't a pool file specified in the CABLE namelist but that's reaching parts of CABLE I don't know much about. That said, the file you list `poolcnpInTumbarumba.csv` seems to be for a single point simulation (I don't know where this file is so I can't check) that isn't suitable for a global simulation so should be fine to remove. We'll see if @MartinDix knows better than me.

**Comment by aidanheerdegen** [comment]
> What do you mean by removing the input files? I guess I don't know what is happening around the ESM1.5 preparations so it's not clear what you mean.

There appears to be a number of input files that just aren't used by the model configuration. So we would like to not include (remove) them from the input directory.

**Comment by dougiesquire** [comment]
I was surprised/curious about all the [`ocean/biogeochemistry` inputs](https://github.com/ACCESS-NRI/access-esm1.5-configs/issues/2#issuecomment-2116892933) so I had a deeper look:

- `cfc_auscom.nc`: I can’t find any evidence that this is used by the ocean (I haven't checked other components)
- `co2_obs.nc`: I presume this is anthropogenic atmospheric co2 concentration. But [`use_access_co2 = .true.` in the `field_table`](https://github.com/ACCESS-NRI/access-esm1.5-configs/blob/94f2bf16b42d672a676354d05d0d914ee122baf9/ocean/field_table#L99) meaning the co2 concentration from the coupler is used. Also, [`aco2_file = INPUT/co2_b35_2D.nc` in the `field_table`](https://github.com/ACCESS-NRI/access-esm1.5-configs/blob/94f2bf16b42d672a676354d05d0d914ee122baf9/ocean/field_table#L110) and this file is not provided as an input. To use the `co2_obs.nc` input for anthropogenic CO2 concentration, this should be `aco2_file = INPUT/co2_obs.nc`. Currently users would get an error if they tried to set `use_access_co2 = .false.`.
- `ocmip2_abiotic_c14_atm_hist_om3_bc-1-9999.nc`: I can’t find any evidence that this is used by the ocean (I haven't checked other components)
- `ocmip2_fice_monthly_om1p5_bc.nc`: this is not used since [`ice_file4gasx = .false.` in the `field_table`](https://github.com/ACCESS-NRI/access-esm1.5-configs/blob/94f2bf16b42d672a676354d05d0d914ee122baf9/ocean/field_table#L101)
- `ocmip2_siple_co2_atm_am2_bc-1-9999.nc`: I can’t find any evidence that this is used by the ocean (I haven't checked other components)
- `ocmip2_xkw_monthly_om1p5_bc.nc`: this is not used since [`gasx_from_file = .false.` in the `field_table`](https://github.com/ACCESS-NRI/access-esm1.5-configs/blob/94f2bf16b42d672a676354d05d0d914ee122baf9/ocean/field_table#L100)

So...

If we only want to keep inputs that are used by the configuration without modification:

- Remove `cfc_auscom.nc`
- Remove `co2_obs.nc`
- Remove `ocmip2_abiotic_c14_atm_hist_om3_bc-1-9999.nc`
- Remove `ocmip2_fice_monthly_om1p5_bc.nc`
- Remove `ocmip2_siple_co2_atm_am2_bc-1-9999.nc`
- Remove `ocmip2_xkw_monthly_om1p5_bc.nc`

or, if we want to support users changing the WOMBAT `field_table` flags `use_access_co2`, `gasx_from_file` and `ice_file4gasx`:

- Remove `cfc_auscom.nc`
- Set [`aco2_file = INPUT/co2_obs.nc` in the `field_table`](https://github.com/ACCESS-NRI/access-esm1.5-configs/blob/94f2bf16b42d672a676354d05d0d914ee122baf9/ocean/field_table#L110)
- Remove `ocmip2_abiotic_c14_atm_hist_om3_bc-1-9999.nc`
- Remove `ocmip2_siple_co2_atm_am2_bc-1-9999.nc`

(Note, someone who knows about these inputs should check all of this)

**Comment by blimlim** [comment]
I've had a go at removing the extra inputs from an esm1.5 configuration – most of the files could be removed without any issues, apart from: 
- `ocmip2_xkw_monthly_om1p5_bc.nc`
- `ocmip2_fice_monthly_om1p5_bc.nc`

The model crashes when either of these are removed even though the related namelist settings are `.false.`. Changing both files' values to zero doesn't seem impact the simulation at all, so it looks like the model is just trying to read in their values without using them.

I'm wondering what a good way to categorise these files in the [restructured input directories](https://github.com/ACCESS-NRI/access-esm1.5-configs/issues/2) be would be. E.g. do you think it would it be more confusing to classify them as `unused` but still require them in the `config.yaml` file, or to instead include them with the other required files despite them being ignored in the config's namelist settings?

**Comment by aidanheerdegen** [comment]
> The model crashes when either of these are removed even though the related namelist settings are .false..

The version of `csiro_bgc.F90` in the ACCESS-ESM1.5 fork doesn't do the check for `gas_from_file` before trying to open the file.

https://github.com/ACCESS-NRI/MOM5/blob/access-esm1.5/src/mom5/ocean_csiro_bgc/csiro_bgc.F90#L2027-L2029

Arguably then this should be back-ported to the ESM1.5 branch.

If it is just this commit then it may be straightforward to cherry-pick it

https://github.com/ACCESS-NRI/MOM5/commit/3f7b6de237d0872c55c557d6dcb16063c4ec355d

But I'll leave that up to @dougiesquire to make that call.

**Comment by dougiesquire** [comment]
> If it is just this commit then it may be straightforward to cherry-pick it
>
> https://github.com/ACCESS-NRI/MOM5/commit/3f7b6de237d0872c55c557d6dcb16063c4ec355d
>
> But I'll leave that up to @dougiesquire to make that call.

I agree. PR [here](https://github.com/ACCESS-NRI/MOM5/pull/22)

**Comment by dougiesquire** [comment]
> I agree. PR [here](https://github.com/ACCESS-NRI/MOM5/pull/22)

@blimlim, this PR has now been merged into the `access-esm1.5` branch. When you get a chance, could you confirm that with this change you can now run without `ocmip2_xkw_monthly_om1p5_bc.nc` and `ocmip2_fice_monthly_om1p5_bc.nc`?

**Comment by blimlim** [comment]
> When you get a chance, could you confirm that with this change you can now run without `ocmip2_xkw_monthly_om1p5_bc.nc` and `ocmip2_fice_monthly_om1p5_bc.nc`?

Can do! I've [added an issue](https://github.com/ACCESS-NRI/ACCESS-ESM1.5) to the ESM1.5 repository to incorporate these changes into a new MOM5 executable, and will test this out once its is ready. @penguian mentioned that it would be useful to set up automatic issues in https://github.com/ACCESS-NRI/ACCESS-ESM1.5/ that get created whenever there are new commits to the esm1.5 components.

**Comment by aidanheerdegen** [comment]
Closing as I think this is completed, but re-open if there is anything outstanding.
